### PR TITLE
Upgrade Stepup-RA to run on Symfony 6.4 and PHP 8.2

### PR DESCRIPTION
**Pull request by parijke**
Upgrade the RA application to support Symfony 6.4 and PHP 8.2

Many structural changes have been made. For details, have a look at the commit history of this PR. Some highlights include:

- Introduction of PHPStan
- All composer dependencies have been upgraded
- Reorganisation of the package configuration
- Other QA Tooling was updated

### CODE REVIEW

File: src/Surfnet/StepupRa/RaBundle/Command/SearchRecoveryTokensCommand.php [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <?php
 
 /**
- * Copyright 2022 SURFnet bv
+ * Copyright 2014 SURFnet bv
```
Comment (MKodde) [inline src/Surfnet/StepupRa/RaBundle/Command/SearchRecoveryTokensCommand.php:4]: See previous comment.

File: src/Surfnet/StepupRa/RaBundle/Command/SearchSecondFactorAuditLogCommand.php
Comment (MKodde) [inline src/Surfnet/StepupRa/RaBundle/Command/SearchSecondFactorAuditLogCommand.php:1]: I see you chmodded quite a lot of files from mask: 644 to 755. Is there a specific reason for this? If not, you might want to restore that?

File: src/Surfnet/StepupRa/RaBundle/Controller/ProfileController.php [hunk 2]
```diff
@@ -18,27 +18,32 @@
...
-        $logger = $this->get('logger');
-
-        $logger->notice('Opening profile page');
-
-        $identity = $this->getUser();
-        $profile = $this->getProfileService()->findByIdentityId($identity->id);
-        return $this->render('SurfnetStepupRaRaBundle:profile:profile.html.twig', ['profile' => $profile]);
+    public function __construct(
+        private readonly ProfileService $profileService,
+        private readonly LoggerInterface $logger,
```
Comment (MKodde) [inline src/Surfnet/StepupRa/RaBundle/Controller/ProfileController.php:31]: :bow: Bye Bye `$this->get` 'in all the things from the service locator.

**Review by MKodde — CHANGES_REQUESTED** [review]
Hi Paul, I absolutely love the work you put in here. Don't be discouraged by my numerous feedback points, they are mostly on the verge of nitpicking.

**Non-code review points**
Please also review these non-code review points;

1. You created this PR from `master`, Not a bad idea at all. We however have some new code on `develop`. I'd say, try to rebase on `develop` to get those changes. Or cherry pick those changes after merging this to master. But I much prefer a rebase onto `develop`.
2. After merging, we need a new `release/6.0` branch based on the develop branch with your changes
3. Next, we need to merge the `release/6.0` branch to master
4. Finally develop can be removed. Master can be renamed to `main`, `main` can be the default branch from that point on.
5. A lot of PHPStan warnings ended up in the baseline. I trust you did a best effort attemt to fix the issues when possible. 
6. The `config/packages/[dev|prod|smoketest|test]` folders should be removed from the project after migrating all env specific config. I see that the web_profiler settings are not yet moved to the `framework.yaml`
7. I see quite some unused imports in the project. PHPStorm can super easily remove them for you. In the `project` pane, right click on the `src` folder: Click `Optimize imports`. Bob's your uncle.
8. Please move the `src/**/Resources/*` to the /config folder. That way we can finally part with that SF 3 convention.
9. The `app` folder can be removed now? If you are not certain please run this past @pmeulen or @phavekes.

**Next action**
My next action is to use the application in the browser. I'm going to check if I can find some regressions there.

File: .github/workflows/test-integration.yml [hunk 3]
```diff
@@ -1,48 +1,26 @@
...
-      - name: Output log files on failure
-        if: failure()
-        run: cd ci/docker && docker-compose exec -T stepup-ra -c 'cat var/log/webtest.log'
+        uses: actions/checkout@v4
+
+      - name: Composer install
+        run: composer install
+
+      - name: Yarn install
+        run: yarn
```
[the remark this answers is not in the document]
Comment (parijke) [reply .github/workflows/test-integration.yml:23]: Done

File: .scrutinizer.yml [hunk 4]
```diff
@@ -1,3 +1,9 @@
+build:
+    environment:
+        php:
+            version: 8.2.12
+#        node: v20.5.0
```
[the remark this answers is not in the document]
Comment (parijke) [reply .scrutinizer.yml:5]: @MKodde maybe someone who knows how to configure scrutinizer can check this? It seems to fail always on node version

File: assets/images/header-logo.png
[the remark this answers is not in the document]
Comment (parijke) [reply assets/images/header-logo.png:1]: Done

File: config/routes/dev/jms_translation.yaml [hunk 5]
```diff
@@ -1,4 +1,4 @@
-JMSTranslationBundle_ui:
-  resource: "@JMSTranslationBundle/Controller/"
-  type:     annotation
-  prefix:   /_trans
+#JMSTranslationBundle_ui:
+#  resource: "@JMSTranslationBundle/Controller/"
+#  type:     attribute
+#  prefix:   /_trans
```
[the remark this answers is not in the document]
Comment (parijke) [reply config/routes/dev/jms_translation.yaml:4]: No, that is undecided

File: src/Surfnet/StepupRa/RaBundle/Command/RevokeRecoveryTokenCommand.php [hunk 6]
```diff
@@ -1,7 +1,7 @@
 <?php
 
 /**
- * Copyright 2022 SURFnet bv
+ * Copyright 2014 SURFnet bv
```
[the remark this answers is not in the document]
Comment (parijke) [reply src/Surfnet/StepupRa/RaBundle/Command/RevokeRecoveryTokenCommand.php:4]: You are absolutely right. The --follow was doing it wrong. Corrected the feature. Maybe we should create some central point for our quality tools and configurations to share across repos

File: src/Surfnet/StepupRa/RaBundle/Command/SearchSecondFactorAuditLogCommand.php
Comment (parijke) [reply src/Surfnet/StepupRa/RaBundle/Command/SearchSecondFactorAuditLogCommand.php:1]: For permission in development.

**Review by parijke — COMMENTED** [review]
Some topics need to be discussed

**Comment by parijke** [comment]
> 5\. A lot of PHPStan warnings ended up in the baseline. I trust you did a best effort attemt to fix the issues when possible.

No I did not... I was in the assumption that a baseline is set to avoid overwhelming changes on an existing code base. Maybe we can do a better job, but the question is: how far....?

File: bin/console [hunk 7]
```diff
@@ -1,42 +1,18 @@
 #!/usr/bin/env php
 <?php
 
-use App\Kernel;
+use src\Kernel;
```
[the remark this answers is not in the document]
Comment (MKodde) [reply bin/console:4]: Fixed this myself

File: config/openconext/parameters.yaml.dist [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+    loa_required_for_login: 'http://dev.openconext.local/assurance/loa3'
+    authentication_context_class_ref: '%loa_required_for_login%'
```
Comment (phavekes) [inline config/openconext/parameters.yaml.dist:43]: This seems to be an new config parameter, and probably does the same as `loa_required_for_login`. Please use the old parameter and drop this, or explain the purpose of this new parameter

**Review by phavekes — CHANGES_REQUESTED** [review]

File: config/openconext/parameters.yaml.dist (hunk 8 above)
Comment (MKodde) [reply config/openconext/parameters.yaml.dist]: Fair point. Merged them. And kept `authentication_context_class_ref

**Review by MKodde — APPROVED** [review]

**Pull request merged**